Tests/TracingMacrosTests/TracedTests.swift

```diff
@@ -0,0 +1,540 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Distributed Tracing open source project
+//
+// Copyright (c) 2020-2024 Apple Inc. and the Swift Distributed Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Distributed Tracing project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntaxMacrosTestSupport
+import Tracing
+import TracingMacros
+import TracingMacrosImplementation
+import XCTest
+
+final class TracedMacroTests: XCTestCase {
+    #if compiler(>=6.0)
+    func test_tracedMacro_requires_body() {
+        assertMacroExpansion(
+            """
+            @Traced
+            func funcWithoutBody()
+            """,
+            expandedSource: """
+                func funcWithoutBody()
+                """,
+            diagnostics: [
+                .init(message: "expected a function with a body", line: 1, column: 1)
+            ],
+            macros: ["Traced": TracedMacro.self]
+        )
+    }
+
+    func test_tracedMacro_sync_nothrow() {
+        assertMacroExpansion(
+            """
+            @Traced
+            func syncNonthrowingExample(param: Int) {
+                print(param)
+            }
+            """,
+            expandedSource: """
+                func syncNonthrowingExample(param: Int) {
+                    withSpan("syncNonthrowingExample") { span in
+                        print(param)
+                    }
+                }
+                """,
+            macros: ["Traced": TracedMacro.self]
+        )
+    }
+
+    func test_tracedMacro_sync_throws() {
+        assertMacroExpansion(
+            """
+            @Traced
+            func syncThrowingExample(param: Int) throws {
```

Comment:
same with async/await

Reply:
Nope—this is what motivated doing the extra work to copy over the effect specifier on the closure instead of leaving it inferred. See the commit message on https://github.com/apple/swift-distributed-tracing-extras/pull/42/commits/3951ba29f71ef03e0fa214af7475d11474bd4bef for a discussion of that.